crates/bls-gadgets/Cargo.toml

```diff
@@ -7,9 +7,9 @@ edition = "2018"
 [dependencies]
 bls-crypto = { path = "../bls-crypto", default-features = false }
 
-algebra = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "sw6", "edwards_sw6", "edwards_bls12", "parallel"] }
+algebra = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "bw6_761", "ed_on_bw6_761", "ed_on_bls12_377", "parallel"] }
 r1cs-core = { git = "https://github.com/scipr-lab/zexe", default-features = false }
-r1cs-std = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "edwards_sw6", "parallel"] }
+r1cs-std = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "ed_on_cp6_782", "parallel"] }
```

Comment:
```suggestion
r1cs-std = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "ed_on_bw6_761", "parallel"] }
```

Reply:
There's some issue with Zexe for this:

```patch
diff --git i/crates/bls-gadgets/Cargo.toml w/crates/bls-gadgets/Cargo.toml
index 21ca060..71019c8 100644
--- i/crates/bls-gadgets/Cargo.toml
+++ w/crates/bls-gadgets/Cargo.toml
@@ -9,7 +9,7 @@ bls-crypto = { path = "../bls-crypto", default-features = false }
 
 algebra = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "bw6_761", "ed_on_bw6_761", "ed_on_bls12_377", "parallel"] }
 r1cs-core = { git = "https://github.com/scipr-lab/zexe", default-features = false }
-r1cs-std = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "ed_on_cp6_782", "parallel"] }
+r1cs-std = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["bls12_377", "ed_on_bw6_761", "parallel"] }
 crypto-primitives = { git = "https://github.com/scipr-lab/zexe", default-features = false, features = ["parallel"] }
 
 # used only when exporting our test helpers to be used in the snark crate
diff --git i/crates/bls-gadgets/src/hash_to_group.rs w/crates/bls-gadgets/src/hash_to_group.rs
index c468c00..4bc1dc4 100644
--- i/crates/bls-gadgets/src/hash_to_group.rs
+++ w/crates/bls-gadgets/src/hash_to_group.rs
@@ -32,7 +32,7 @@ use crypto_primitives::{
 };
 use r1cs_core::{ConstraintSystem, SynthesisError};
 use r1cs_std::{
-    alloc::AllocGadget, bits::ToBitsGadget, boolean::Boolean, ed_on_cp6_782::EdwardsGadget,
+    alloc::AllocGadget, bits::ToBitsGadget, boolean::Boolean, ed_on_bw6_761::EdwardsGadget,
     groups::bls12::G1Gadget, groups::GroupGadget, uint8::UInt8, Assignment,
 };
 use std::{borrow::Borrow, marker::PhantomData};
```

Results in this error:

```
error[E0432]: unresolved import `r1cs_std::ed_on_bw6_761`
  --> crates/bls-gadgets/src/hash_to_group.rs:35:63
   |
35 |     alloc::AllocGadget, bits::ToBitsGadget, boolean::Boolean, ed_on_bw6_761::EdwardsGadget,
   |                                                               ^^^^^^^^^^^^^ could not find `ed_on_bw6_761` in `r1cs_std`
```